src/actions.js

```diff
@@ -28,5 +41,8 @@ export const addCategory = (categoryName) => (dispatch) => {
         .ref('/categories')
         .push()
         .set(categoryName)
-        .then(() => dispatch(setCreatingState(false)));
-}
+        .then(() => {
```

Comment:
1. Spróbuj zrobic funkcje _addCategory_ jako asynchroniczną: `export const addCategory = (categoryName) => async (dispatch) => { ... }`

Reply:
Dodatkowo jak chcesz złapać error możesz otoczyć to w try/catch.